src/vs/platform/terminal/common/capabilities/commandDetectionCapability.ts

```diff
@@ -109,6 +109,8 @@ export class CommandDetectionCapability extends Disposable implements ICommandDe
 	readonly onBeforeCommandFinished = this._onBeforeCommandFinished.event;
 	private readonly _onCommandFinished = this._register(new Emitter<ITerminalCommand>());
 	readonly onCommandFinished = this._onCommandFinished.event;
+	private readonly _onPreCommandFinishedWindows = this._register(new Emitter<void>());
+	readonly onPreCommandFinishedWindows = this._onPreCommandFinishedWindows.event;
```

Comment:
Could we just use `onCommandFinished` or `onBeforeCommandFinished` instead?

Reply:
I tried that originally - the only issue is then if you start a long running command, it never opens it.

Reply:
Maybe I'm reading it wrong, but isn't this new event fired synchronously in the same task as both onCommandFinished and onBeforeCommandFinished?

Reply:
Regardless, the main feedback I wanted to give here is that we want to keep Windows edge cases encapsulated and fake it/polyfill so it essentially acts the same on all platforms. So instead of a new event, firing the actual `_onCommandExecuted` when it's detected not to have gone off and in is a windows pty.

Reply:
oh ok, makes sense